config/config.go

```diff
@@ -201,19 +198,25 @@ func Validate() error {
 	if maxMessageSize <= 0 || maxMessageSize > 2048 {
 		return errors.New(" max_message_size_in_mb cannot be more than 2GB (i.e 2048 MB) and has to be a positive number")
 	}
-    if err = curationChecks() ;err !=nil {
-    	return err
+	if err = allowedUserConfigurationChecks(); err != nil {
+		return err
 	}
 	return validateMeteringChecks()
 }
-func curationChecks() (error) {
-	if GetBool(IsCurationInProgress) && GetString(BlockChainNetworkSelected)=="main" {
-		return fmt.Errorf("service cannot be curated while set up against Ethereum mainnet,the flag %v is set to true",IsCurationInProgress)
-	}
-	if GetBool(IsCurationInProgress) && !common.IsHexAddress(GetString(CurationAddressForValidation)) {
-		return fmt.Errorf("a valid Address needs to be specified for the config %v to ensure that, only this user can make calls",CurationAddressForValidation)
-	}
 
+//Feature in Daemon to restrict access to only certain users , this feature is useful,when you are
+//in a test environment and dont want everyone to make requests to your service.
+//Since this was flag was introduced to restrict users while in testing mode, we dont want this configuration
+//to be mistakenly set on mainnet
+func allowedUserConfigurationChecks() error {
+	if GetBool(AllowedUserFlag) {
```

Comment:
With the above comment, we can get the condition with the allowedUserNames itself...

Reply:
we wanted to use a flag to allow  a feature to be used, rather than derive the behaviour based on a field being populated , the code looks slightly better with the flag checks, however agree we can derive if the feature is On/off based on the value of the allowedUsers configuration